### DISCUSSION TITLE
Using custom search-box webpart to replace the PnP search-box

### DISCUSSION THREAD

**Discussion by hsamdani12** [microsoft-search/pnp-modern-search#2479] [Q&A]
Hi,
Background - I've 100s of Client SharePoint Sites and Project SharePoint Sites. Every Client Sites have related 4-5 Project Sites.
I've a requirement where I need to dynamically build a KQL and append to the keyword the user typed in the search box.
For eg: If the user types "proposal", I want to append Path: ProjectSiteUrlA Path: ProjectSiteUrlB Path: ProjectSiteUrlC.  I would be getting these ProjectSiteUrlA, B and C. by calling an API from another system. Basically the search box webpart and search result webpart would need to be deployed to 100's of Client SharePoint site and from the Client site I want to get the Project Sites URL by calling the API and then give the ability for the use to search with a keyword scoped to only the Project Sites related to that Client.

So I need to replace the following webpart with custom one and make it work with the search result webpart -
https://microsoft-search.github.io/pnp-modern-search/usage/search-box

Is there a way for me to use web component to customize the behaviour of the PnP Search box so I don't need to build a new webaprt ?

Would appreciate any feedback about the approach I can take for this ?

**Comment by wobba** [comment]
@hsamdani12 there is work happening to support query transformations. See https://github.com/microsoft-search/pnp-modern-search/pull/2450 and https://github.com/microsoft-search/pnp-modern-search/pull/2035

**Comment by hsamdani12** [comment]
@wobba thanks, it would be great to have the query transformation.
But just needed to know if I were to build a custom search webpart and pass the dynamically built KQL string to PnP search result webpart using dynamic data feature would it be valid solution/approach in the scenario mentioned ?
https://learn.microsoft.com/en-us/sharepoint/dev/spfx/dynamic-data

**Reply by wobba** [reply]
Sure, that would work as well. You're likely better off waiting for a supported feature to stay in sync, but a custom search box web part should work just as well.

**Comment by hsamdani12** [comment]
@wobba  Thanks for your suggestion to wait and use the supported feature of query transformation.  I beleive in that case would it possible to do this just by using configuration changes in the PnP search box and PnP search result webparts OR do I still need to write code in web component - https://microsoft-search.github.io/pnp-modern-search/extensibility/custom_web_component/

**Reply by wobba** [reply]
Look at the PR's and see if they solve it for you, or if you need to go custom.

**Reply by hsamdani12** [reply]
@wobba ok thanks.
